pkg/operator/controllers/checker/unsupportconditionchecker.go

```diff
@@ -0,0 +1,118 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	consolev1 "github.com/openshift/api/console/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	consoleclient "github.com/openshift/client-go/console/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	clusterOperatorName = "aro"
+	reasonAsExpected    = "AsExpected"
+)
+
+type NodesChecker struct {
+	kubernetescli kubernetes.Interface
+	consolecli    consoleclient.Interface
+	configcli     configclient.Interface
+}
+
+func NewNodesChecker(kubernetescli kubernetes.Interface, consolecli consoleclient.Interface, configcli configclient.Interface) *NodesChecker {
+	return &NodesChecker{
+		kubernetescli: kubernetescli,
+		consolecli:    consolecli,
+		configcli:     configcli,
+	}
+}
+
+func (nc *NodesChecker) Name() string {
+	return "WorkerNodesCountChecker"
+}
+
+func (nc *NodesChecker) Check(ctx context.Context) error {
+	bannerName := "aro-sre"
+	nodes, err := nc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	co, _ := nc.getOrCreateClusterOperator(ctx)
+
+	currentTime := metav1.Now()
+	condition := []configv1.ClusterOperatorStatusCondition{
+		{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionFalse,
+			LastTransitionTime: currentTime,
+			Reason:             reasonAsExpected,
+		},
+		{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: currentTime,
+			Reason:             reasonAsExpected,
+		},
+	}
+
+	if len(nodes.Items) < 3 {
+		// nc.setClusterOperatorDisabled(ctx, co)
+		v1helpers.SetStatusCondition(&co.Status.Conditions, condition[0])
+
+		banner := nc.newBanner(bannerName, "The number of the current worker nodes are less then 3")
+
+		_, err := nc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, bannerName, metav1.GetOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+
+		// if the object doesn't exist Create
+		if err != nil && kerrors.IsNotFound(err) {
+			_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, banner, metav1.CreateOptions{})
+			return err
+		}
+
+		// if there's no errors, object found then update
+		_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, banner, metav1.UpdateOptions{})
+		return err
+	} else {
+		v1helpers.SetStatusCondition(&co.Status.Conditions, condition[1])
+
+		err = nc.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, bannerName, metav1.DeleteOptions{})
+		return err
+	}
+}
```

Comment:
I know this is still a draft PR, but this function is a little long, if you split it smaller chunks it will be easier to read, and more importantly, easier to test

Reply:
@facchettos - Thanks Jeremy for your time to review and sorry for the late response, will update accordingly.